### PR TITLE
[expo-battery] Add NOT_CHARGING and fix Android battery status mapping (#44212)

### PR DESCRIPTION
**Pull request by vj2303**
## Summary

Fixes incorrect `BatteryState.UNPLUGGED` when Android reports [`BATTERY_STATUS_NOT_CHARGING`](https://developer.android.com/reference/android/os/BatteryManager#BATTERY_STATUS_NOT_CHARGING) while power is still connected (e.g. battery protection capping charge at 80%). Reported in #44212.

- **`BATTERY_STATUS_DISCHARGING`** → `UNPLUGGED` (discharging on battery).
- **`BATTERY_STATUS_NOT_CHARGING`** + **`EXTRA_PLUGGED` ≠ 0** → new **`BatteryState.NOT_CHARGING`**.
- **`BATTERY_STATUS_NOT_CHARGING`** + unplugged → **`UNPLUGGED`** (fallback for OEM edge cases).

## Test plan

- [ ] CI
- [ ] Android: `getBatteryStateAsync` / `addBatteryStateListener` with a device or emulator that reports `NOT_CHARGING` while plugged (optional)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: packages/expo-battery/src/Battery.types.ts [hunk 1]
```diff
@@ -32,6 +33,12 @@ export enum BatteryState {
    * If the battery level is full.
    */
   FULL,
+  /**
+   * **Android only.** The battery is not charging while power is connected (AC/USB/wireless), for
+   * example when battery protection limits charge to 80%, or optimized charging pauses. This
+   * differs from `UNPLUGGED` (discharging on battery). On iOS and web, this value is never returned.
+   */
```
Comment (behenate) [inline packages/expo-battery/src/Battery.types.ts]:
```suggestion
   * The battery is not charging while power is connected (AC/USB/wireless), for
   * example when battery protection limits charge to 80%, or optimized charging pauses. This
   * differs from `UNPLUGGED` (discharging on battery). On iOS and web, this value is never returned.
   * 
   * @platform android
   */
```

Let's use the platform annotation here

**Review by behenate — CHANGES_REQUESTED** [review]
Looks good, I've left a small suggestion 😄

**Comment by vj2303** [comment]
Thanks @behenate — good call on using the platform tag. I removed the “Android only” wording from the doc block and added @platform android so it lines up with the rest of the docs. Rebuilt expo-battery and pushed; ready for another look when you have time.

**Review by behenate — APPROVED** [review]

**Review by vonovak — APPROVED** [review]

**Comment by vonovak** [comment]
@vj2303 one more thing, could you regenerate the docs to have them updated for this change?

just run `et gdad -p expo-battery` from repo root

thank you

File: packages/expo-battery/src/Battery.types.ts [hunk 2]
```diff
@@ -32,6 +33,14 @@ export enum BatteryState {
...
    */
   FULL,
+  /**
+   * The battery is not charging while power is connected (AC/USB/wireless), for
+   * example when battery protection limits charge to 80%, or optimized charging pauses. This
+   * differs from `UNPLUGGED` (discharging on battery). On iOS and web, this value is never returned.
+   *
+   * @platform android
+   */
+  NOT_CHARGING,
```
Comment (vonovak) [inline packages/expo-battery/src/Battery.types.ts:43]: [🏗️ **design**] This adds a new public enum value which changes runtime behavior: Android users who previously received `UNPLUGGED` when plugged-in-but-not-charging will now get `NOT_CHARGING`. Any consumer code with exhaustive switch/case on `BatteryState` (without a default) will hit a new unhandled case - that's okay but means we likely won't pick it for sdk 55.

Also, `packages/expo-battery/src/Battery.ts` lines 47 and 157 reference "four states" in the JSDoc — now stale with five values. Worth updating in this PR.

**Review by vonovak — CHANGES_REQUESTED** [review]
**Suggested verdict: 🔴 REQUEST_CHANGES**

_🤖 This is an automated review. Addressing it doesn't guarantee a merge._

The fix correctly distinguishes `BATTERY_STATUS_NOT_CHARGING` (plugged but not charging, e.g. battery protection at 80%) from `BATTERY_STATUS_DISCHARGING` (on battery). The Android mapping logic with the `EXTRA_PLUGGED` check, new enum value, and TS docs are sound. iOS and web are unaffected since they can never produce value 4.

Two things need updating outside the diff:

1. **`apps/test-suite/tests/Battery.js:37-41`** — The test hardcodes `[CHARGING, FULL, UNPLUGGED]` as valid battery states. On an Android device that reports `NOT_CHARGING` (e.g. plugged in with battery protection), `getBatteryStateAsync()` will return `4` and the test will fail. Add `Battery.BatteryState.NOT_CHARGING` to the accepted values.

2. **`apps/native-component-list/src/screens/BatteryScreen.tsx:67-79`** — The `getBatteryStateString` switch falls through to `'UNKNOWN'` for the new state. Since this is the demo screen for the module, add a `case Battery.BatteryState.NOT_CHARGING: return 'NOT_CHARGING';` case.

**Review by vonovak — APPROVED** [review]
